src/main/java/com/scaleunlimited/flinkcrawler/functions/CheckUrlWithRobotsFunction.java

```diff
@@ -109,24 +115,37 @@ public void run() {
                     }
                 }
 
+                int httpStatusCode;
                 long robotsFetchRetryDelay;
                 try {
                     FetchedResult result = _fetcher.get(robotsUrl);
-                    LOGGER.trace("CheckUrlWithRobotsFunction fetched URL '{}' with status {}", robotsUrl,
-                            result.getStatusCode());
-                    robotsFetchRetryDelay = calcRobotsFetchRetryDelay(result.getStatusCode());
-                    if (result.getStatusCode() != HttpStatus.SC_OK) {
-                        rules = _parser.failedFetch(result.getStatusCode());
+                    httpStatusCode = result.getStatusCode();
+                    LOGGER.trace("CheckUrlWithRobotsFunction fetched URL '{}' with status {}",
+                            robotsUrl, httpStatusCode);
+                    robotsFetchRetryDelay = calcRobotsFetchRetryDelay(httpStatusCode);
+                    if (httpStatusCode != HttpStatus.SC_OK) {
+                        rules = _parser.failedFetch(httpStatusCode);
+                        if ((httpStatusCode >= 400) && (httpStatusCode < 500)) {
+                            _numMissingRobots++;
+                        } else {
+                            _numFailedRobots++;
+                        }
                     } else {
                         rules = _parser.parseContent(robotsUrl, result.getContent(),
                                 result.getContentType(), _fetcher.getUserAgent().getAgentName());
+                        _numFetchedRobots++;
                     }
                 } catch (Exception e) {
-                    robotsFetchRetryDelay = calcRobotsFetchRetryDelay(
-                            HttpStatus.SC_INTERNAL_SERVER_ERROR);
-                    rules = _parser.failedFetch(HttpStatus.SC_INTERNAL_SERVER_ERROR);
+                    httpStatusCode = HttpStatus.SC_INTERNAL_SERVER_ERROR;
+                    robotsFetchRetryDelay = calcRobotsFetchRetryDelay(httpStatusCode);
+                    rules = _parser.failedFetch(httpStatusCode);
+                    _numFailedRobots++;
                 }
 
+                String hostname = url.getHostname();
+                if (!hostname.equals(_currentDomain)) {
+                    printCounters(hostname, httpStatusCode);
```

Comment:
Print counters if _rules.put() says this is something new. So no need for _currentDomain